Hybrid-state.md

```diff
@@ -0,0 +1,93 @@
+
+* If the suite is migrated to webdriverIO then make the changes in webdriverIO.
+  
+* If the suite is still in protractor then make the changes in protractor.
+
+  **Edge Case**: If we have a migration PR open for a particular suite say _Suite A_ and at the same time any other contributor also made a PR with some changes in _Suite A_ then, we need to sync up the changes of both the PR. So, if the migration PR get's merged first, then the other contributor needs to migrate his/her changes to webdriverIO (contributor can take help from [Shivam Jha](#contact)). Otherwise, migration PR needs to be updated with the changes.
+
+  Also, if migration PR will get merged first then everyone will be notified about it accordingly as migration PR will have the **PR: that requires post-merge sync to the HEAD label.**
+
+#### If contributor wants to make some changes to the common files
+
+  The tests might break if the changes are only applied in one version of the common file. So we need to make sure to keep the two versions of the common file in sync. We will not merge the PR of the contributor if it's not synced. The contributor can take help from [Shivam Jha](#contact) or  [Guide to migrate e2e tests](Guide-to-migrate-e2e-tests.md) in order to make the necessary changes for the other version.
+  
+#### Current approach to migrate the dependencies
+
+To reduce the duplicate code in the codebase we are going to migrate only those portions of the dependencies which are used in the suite that we are migrating. The remaining portions will be migrated when a suite that depends upon them is being migrated.
+
+Let's understand this situation with a simple test case.
+
+We have two test files:
+
+|**File**            | **Dependencies**         |
+|--------------------|--------------------------|
+|AdditonalPlayer     |ExplorationEditorPage, ExplorationPlayerPage, LibraryPage|
+|AdditonalEditorFeaturesModals| ExplorationEditorPage |
+
+* First, we will migrate **AdditionalPlayer** and its dependencies **ExplorationPlayerPage**, **LibraryPage**, and **ExplorationEditorPage**. As **ExplorationPlayerPage** and **LibraryPage** doesn't have any suite left that are dependent on them, we will delete their protractor version. But for **ExplorationEditorPage**  we still have one dependent suite i.e. **AdditonalEditorFeatures**, so we cannot delete the protractor version (as a protractor suite cannot work on webdriverIO util files).
+
+* Now we will migrate the **AdditonalPlayer** suite and after its migration, we can delete the protractor version of **ExplorationEditorPage**.
+
+ **Note**: **ExplorationEditorPage** file will be a common file during the migration phase (presented in both webdriverIO and protractor version). To reduce code duplication to some extent we will only migrate the portion of ExplorationEditorPage that is being used in the **AdditonalPlayer** suite, rest portion can be migrated when we will migrate **AdditonalEditorFeaturesModals**.
```

Comment:
```suggestion
 **Note**: **ExplorationEditorPage** file will be a common file during the migration phase (presented in both webdriverIO and protractor version). To reduce code duplication to some extent we will only migrate the portion of ExplorationEditorPage that is being used in the **AdditonalPlayer** suite. The remaining portion can be migrated when we migrate **AdditonalEditorFeaturesModals**.
```

This also needs to be indented one more space

Reply:
Done.